dhall/src/Dhall/Parser/Token.hs

```diff
@@ -197,10 +200,10 @@ doubleLiteral = (do
 -}
 doubleInfinity :: Parser Double
 doubleInfinity = (do
-    let negative = fmap (\_ -> negate) (Text.Parser.Char.char '-')
+    let negative = fmap (\_ -> negate) (char '-')
     sign <- negative <|> pure id
-    a <- Text.Parser.Char.text "Infinity" >> whitespace >> return (1.0/0.0)
-    return (sign a) ) <?> "double infinity"
+    a <- text "Infinity" >> whitespace >> return (1.0/0.0)
+    return (sign a) ) <?> "literal"
```

Comment:
It seems like a bit of "layering violation" (not sure if that's the right name) to assign this specific parser the name "literal".

I wonder whether we could instead assign these names at the `primitiveExpression` level:

https://github.com/dhall-lang/dhall-haskell/blob/ad443cd6851af215111e21d8ac92a9fb3cda7403/dhall/src/Dhall/Parser/Expression.hs#L354-L372

Could we do something like this?

```haskell
primitiveExpression =
    noted
        (   (choice [alternative00, alternative01, alternative02] <?> "literal)
        <|> …

Reply:
@sjakobi: It's mainly for ease of maintenance so that we don't have to take care that all uses of `builtin`/`operator`/`keyword` in `Expression.hs` are properly annotated.

For `operator`/`built-in` you could limit the annotations to just one place in the code in `Expression.hs`, but occurrences of `keyword` are spread out all over `Expression.hs`, so I figured it was easier to just upstream the annotations into `Token.hs`

Also, consider that if we later decide to refactor `Expression.hs`, that refactor will be less error-prone if we don't need to ensure that annotations are properly maintained.

Reply:
That's a good point! :+1: